File: 프로그래머스/unrated/161989. 덧칠하기/덧칠하기.py
def solution(n, m, section):
    tmp = section[0]
    i = 1
    result = 1
    while i < len(section):
        if section[i] - tmp >= m:
            tmp = section[i]
            result += 1
        i += 1
    
    return result
            
            
        
        
        
        

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.java
class Solution {
    public int solution(int n, int m, int[] section) {
        int result = 1;
        int start = section[0];
        for (int i=0; i<section.length; i++) {
            if (section[i] - start + 1 > m) {
                result += 1;
                start = section[i];
            }
        }
        return result;
    }
}